aws4.js

```diff
@@ -105,8 +105,8 @@ RequestSigner.prototype.sign = function() {
       if (this.service === 's3')
         headers['X-Amz-Content-Sha256'] = hash(this.request.body || '', 'hex')
 
-      if (headers['X-Amz-Date'])
-        this.datetime = headers['X-Amz-Date']
+      if (headers['X-Amz-Date'] || headers['x-amz-date'] || headers['date'] || headers['Date'])
+        this.datetime = headers['X-Amz-Date'] || headers['x-amz-date'] || headers['date'] || headers['Date']
```

Comment:
`Date` and `date` (both variations) are already covered by the `getDateTime` function below – see line 125